apps/etl/extraction/sources/base/handler.py

```diff
@@ -20,38 +21,94 @@ class ValidationResponse(typing.TypedDict):
     status: str
 
 
-class BaseExtraction:
+class BaseExtraction(abc.ABC):
     """
     Handles data extraction.
     """
 
-    @classmethod
+    @abc.abstractmethod
+    def extract(extraction_object: ExtractionData) -> int:
+        """
+        Not Implemented
+        return extraction id
+        """
+        raise NotImplementedError()
+
+    def get_file_extension(self, source=None):
+        source_file_extension_map = {
+            ExtractionData.Source.DESINVENTAR: ".zip",
+            ExtractionData.Source.IBTRACS: ".csv",
+        }
+        return source_file_extension_map.get(source, ".json")
+
+    def get_headers(self, source):
+        source_headers_map = {
+            ExtractionData.Source.EMDAT: {"Authorization": etl_config.EMDAT_AUTHORIZATION_KEY},
+            ExtractionData.Source.DESINVENTAR: None,
+            ExtractionData.Source.GFD: None,
+            ExtractionData.Source.IBTRACS: None,
+            ExtractionData.Source.USGS: None,
+            # TODO
+            ExtractionData.Source.PDC: {"Authorization": "Bearer {}".format(etl_config.PDC_SENTRY_AUTHORIZATION_KEY)},
+        }
+        return source_headers_map.get(source, {"accept": "application/json"})
```

Comment:
This should not be defined on base extraction

Reply:
Done